hudi-hadoop-common/src/main/java/org/apache/hudi/common/util/HFileUtils.java

```diff
@@ -199,26 +199,26 @@ public byte[] serializeRecordsToLogBlock(HoodieStorage storage,
       }
 
       final byte[] recordBytes = serializeRecord(record, writerSchema, keyFieldName);
-      // If key exists in the map, append to its list. If not, create a new list.
-      // Get the existing list of recordBytes for the recordKey, or an empty list if it doesn't exist
-      List<byte[]> recordBytesList = sortedRecordsMap.getOrDefault(recordKey, new ArrayList<>());
-      recordBytesList.add(recordBytes);
-      // Put the updated list back into the map
-      sortedRecordsMap.put(recordKey, recordBytesList);
+      if (sortedRecordsMap.containsKey(recordKey)) {
+        LOG.error("Found duplicate record with recordKey: " + recordKey);
+        printRecord("Previous record", sortedRecordsMap.get(recordKey), writerSchema);
+        printRecord("Current record", recordBytes, writerSchema);
+        throw new HoodieException(String.format("Writing multiple records with same key %s not supported for %s",
+            recordKey, this.getClass().getName()));
```

Comment:
`this.getClass().getName()` will be `HFileUtils`, should this say something else like "for record index"?